v12/storage-dotnet-redundancy-ha-ra-grs/Program.cs

```diff
@@ -0,0 +1,224 @@
+        },
+    };
+
+    // Add an HttpPipelinePolicy for debugging purposes
+    // This will help us see which endpoint each retry request is being sent to
+    options.AddPolicy(new ShowRequestInfoPolicy(), Azure.Core.HttpPipelinePosition.PerRetry);
+    optionsSecondary.AddPolicy(new ShowRequestInfoPolicy(), Azure.Core.HttpPipelinePosition.PerRetry);
+
+    try
+    {
+        // Create a client object for the blob service with the options defined above 
+        BlobServiceClient blobServiceClient = new(primaryAccountUri, new DefaultAzureCredential(), options);
+
+        // Create a client object for the blob service which points to the secondary region endpoint
+        BlobServiceClient blobServiceClientSecondary = new(secondaryAccountUri, new DefaultAzureCredential(), optionsSecondary);
+
+        // Get some statistics about the secondary region when using RA-GRS or RA-GZRS
+        Response<BlobServiceStatistics> responseStats = await blobServiceClientSecondary.GetStatisticsAsync();
+        if (responseStats.Value.GeoReplication.Status == BlobGeoReplicationStatus.Live)
+        {
+            Console.WriteLine("Secondary region is live");
+            Console.WriteLine($"Last sync time: {responseStats.Value.GeoReplication.LastSyncedOn}");
+        }
+
+        CancellationTokenSource source = new CancellationTokenSource();
+        CancellationToken cancellationToken = source.Token;
+
+        // Create a unique name for the container
+        string containerName = $"container-{Guid.NewGuid()}";
+
+        // Create the container and return a container client object
+        Console.WriteLine("\nCreating container");
+        containerClient = await blobServiceClient.CreateBlobContainerAsync(containerName, PublicAccessType.None, null, cancellationToken);
+
+        if (await containerClient.ExistsAsync())
+        {
+            Console.WriteLine($"Created container {containerClient.Name}\n");
+        }
+
+        // Create a new block blob client object
+        // The blob client retains the credential and client options
+        BlobClient blobClient = containerClient.GetBlobClient(blobName);
+
+        // Upload the data
+        Console.WriteLine($"\nUploading blob: {blobName}");
+        await blobClient.UploadAsync(BinaryData.FromString("If at first you don't succeed, hopefully you have a good retry policy.").ToStream(), overwrite: true);
+
+        // Check to see if the blob has been replicated to the secondary data center
+        BlobClient blobSecondary = blobServiceClientSecondary.GetBlobContainerClient(containerName).GetBlobClient(blobName);
+        int counter = 0;
+        Console.WriteLine("Checking secondary region endpoint to see if the blob has replicated");
+        while (counter < 60)
+        {
+            counter++;
+
+            Console.WriteLine("Attempt {0} to see if the blob has replicated", counter);
```

Comment:
You've been using interpolated strings for most of the output. I'd use it here as well:

```suggestion
            Console.WriteLine($"Attempt {counter} to see if the blob has replicated");
```

Reply:
Thanks for the review, @BillWagner! I'll make this change.